sdk/servicebus/azure-servicebus/tests/servicebus_preparer.py

```diff
@@ -22,13 +28,116 @@
+        name = "rgpy-" + name
+        if len(name) > 90:
+            name = name[:90]
+        return name
+
+    def create_resource(self, name, **kwargs):
+        if self.is_live and self._need_creation:
+            self.client = self.create_mgmt_client(ResourceManagementClient, base_url=BASE_URL, credential_scopes=CREDENTIAL_SCOPES)
+            parameters = {"location": self.location}
+            expiry = datetime.datetime.utcnow() + self.delete_after_tag_timedelta
+            parameters["tags"] = {"DeleteAfter": expiry.replace(microsecond=0).isoformat()}
+
+            parameters["tags"]["BuildId"] = os.environ.get("BUILD_BUILDID", "local")
+            parameters["tags"]["BuildJob"] = os.environ.get("AGENT_JOBNAME", "local")
+            parameters["tags"]["BuildNumber"] = os.environ.get("BUILD_BUILDNUMBER", "local")
+            parameters["tags"]["BuildReason"] = os.environ.get("BUILD_REASON", "local")
+            try:
+                # Prefixing all RGs created here with 'rgpy-' for tracing purposes
+                name = self._prefix_name(name)
+                logging.info(
+                    "Attempting to create a Resource Group with name {} and parameters {}".format(name, parameters)
+                )
+                self.resource = self.client.resource_groups.create_or_update(name, parameters)
+            except Exception as ex:
+                if "ReservedResourceName" in str(ex):
+                    raise ReservedResourceNameError(name)
+                raise
+        else:
+            self.resource = self.resource or FakeResource(
+                name=name,
+                id="/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/" + name,
+            )
+        if name != self.moniker:
+            try:
+                self.test_class_instance.scrubber.register_name_pair(name, self.moniker)
+            # tests using the test proxy don't have a scrubber instance
+            except AttributeError:
+                add_general_regex_sanitizer(regex=name, value=self.moniker)
+        return {
+            self.parameter_name: self.resource,
+            self.parameter_name_for_location: self.location,
+        }
+
+    def remove_resource(self, name, **kwargs):
+        if self.is_live and self._need_creation:
+            name = self._prefix_name(name)
+            try:
+                if "wait_timeout" in kwargs:
+                    azure_poller = self.client.resource_groups.begin_delete(name)
+                    azure_poller.wait(kwargs.get("wait_timeout"))
+                    if azure_poller.done():
+                        return
+                    raise AzureTestError("Timed out waiting for resource group to be deleted.")
+                else:
+                    self.client.resource_groups.begin_delete(name, polling=False).result()
+            except Exception as err:  # NOTE: some track 1 libraries do not have azure-core installed. Cannot use HttpResponseError here
```

Comment:
This comment doesn't apply anymore ? We arent testing track 1, so should we add back HttpResponseError :)